packages/flare/python/cron_job_ingest_events.py

```diff
@@ -29,44 +33,133 @@ def main():
         raise Exception(str(e))
 
     app: client.Application = splunk_service.apps[APP_NAME]
-    flare_api = FlareAPI(app=app)
+    create_collection(app=app)
+
+    # To avoid cron jobs from doing the same work at the same time, exit new cron jobs if a cron job is already doing work
+    if get_job_running(app):
+        return
+
+    api_key: Optional[str] = None
+    tenant_id: Optional[int] = None
+    for item in app.service.storage_passwords.list():
+        if item.content.username == "api_key":
+            api_key = item.clear_password
+
+        if item.content.username == "tenant_id":
+            tenant_id = (
+                int(item.clear_password) if item.clear_password is not None else None
+            )
+
+    if not api_key:
+        raise Exception("API key not found")
 
-    from_ = get_from_value(app=app)
-    event_feed = flare_api.retrieve_feed(from_=from_)
-    set_from_value(app=app, next=event_feed["next"])
+    if not tenant_id:
```

Comment:
Do we really want to crash in this case? Should it be fine if the user wants to use his default tenant?

Reply:
You cannot finish the configuration if you don't select a tenant. The UI dropdown will auto select the default tenant

Reply:
ok sure. Personally I would have pushed that to the nice-to-haves.